pypit/arcoadd.py

```diff
@@ -0,0 +1,698 @@
+    ----------
+    spec : XSpectrum1D
+    mask : ndarray, optional
+      Additional input mask with True = masked
+      This needs to have the same size as the masked spectrum
+    nsig : float, optional
+      Clip sigma
+    niter : int, optional
+      Number of clipping iterations
+    **kwargs : optional
+      Passed to each call of sigma_clipped_stats
+
+    Returns
+    -------
+    med_spec, std_spec
+    """
+    from astropy.stats import sigma_clipped_stats
+    #goodpix = WHERE(refivar GT 0.0 AND finite(refflux) AND finite(refivar) $
+    #            AND refmask EQ 1 AND refivar LT 1.0d8)
+    mean_spec, med_spec, std_spec = sigma_clipped_stats(spec.flux, sigma=nsig, iters=niter, **kwargs)
+    # Clip a bit
+    #badpix = np.any([spec.flux.value < 0.5*np.abs(med_spec)])
+    badpix = spec.flux.value < 0.5*np.abs(med_spec)
+    mean_spec, med_spec, std_spec = sigma_clipped_stats(spec.flux.value, mask=badpix,
+                                                        sigma=nsig, iters=niter, **kwargs)
+    # Return
+    return med_spec, std_spec
+
+
+def scale_spectra(spectra, sn2, iref=0, scale_method='auto', hand_scale=None,
+                  SN_MAX_MEDSCALE=2., SN_MIN_MEDSCALE=0.5, **kwargs):
+    """
+    Parameters
+    ----------
+    spectra : XSpectrum1D
+      Rebinned spectra
+      These should be registered, i.e. pixel 0 has the same wavelength for all
+    sn2 : ndarray
+      S/N**2 estimates for each spectrum
+    iref : int, optional
+      Index of reference spectrum
+    scale_method : str, optional
+      Method for scaling
+       'auto' -- Use automatic method based on RMS of S/N
+       'hand' -- Use input scale factors
+       'median' -- Use calcualted median value
+    SN_MIN_MEDSCALE : float, optional
+      Maximum RMS S/N allowed to automatically apply median scaling
+    SN_MAX_MEDSCALE : float, optional
+      Maximum RMS S/N allowed to automatically apply median scaling
+
+    Returns
+    -------
+    scales : list of float or ndarray
+      Scale value (or arrays) applied to the data
+    omethod : str
+      Method applied (mainly useful if auto was adopted)
+       'hand'
+       'median'
+       'none_SN'
+    """
+    # Init
+    med_ref = None
+    rms_sn = np.sqrt(np.mean(sn2)) # Root Mean S/N**2 value for all spectra
+    # Check for wavelength registration
+    gdp = np.all(~spectra.data['flux'].mask, axis=0)
+    gidx = np.where(gdp)[0]
+    if not np.isclose(spectra.data['wave'][0,gidx[0]],spectra.data['wave'][1,gidx[0]]):
+        msgs.error("Input spectra are not registered!")
+    # Loop on exposures
+    scales = []
+    for qq in range(spectra.nspec):
+        if scale_method == 'hand':
+            omethod = 'hand'
+            # Input?
+            if hand_scale is None:
+                msgs.error("Need to provide hand_scale parameter, one value per spectrum")
+            spectra.data['flux'][qq,:] *= hand_scale[qq]
+            spectra.data['sig'][qq,:] /= hand_scale[qq]
```

Comment:
I think you're right that the sig values should be multiplied instead of divided by the scale factor. I checked XIDL's coadder, and they work with ivar arrays there, which they then divide by the hand_scale**2, so it makes sense to me that sig values should also be multiplied by the scale factor.

I've changed this line (and lines 357 and 375).